### ISSUE TITLE
user registration while a wallet is disconnected (but is now up) is not handled well

### ISSUE THREAD

**Issue by jolan** [decred/dcrstakepool#70]
14:23 < Dyrk> DCRS: importScriptFn nonequiv failure 1 on servers 1, 2
14:23 < Dyrk> [INF] DCRS: RPC ImportScript failed on connection error: non equivalent error returned 1
14:23 < Dyrk> [INF] DCRS: Wallet RPC manager shutting down
14:24 < Dyrk> when this happens in the stake pool (may be few times) it stop interacting with wallets
14:25 < Dyrk> only restart helps
14:25 < Dyrk> as I understant this bug was introduced in "Add smarter wallet connectivity " 0881103 right?

I hit this too while testing.  I rebooted a wallet server. The wallet was shown as disconnected but was running again.  When a user registered, I hit the same thing as @Dyrk.  I didn't investigate at all but my guess is that it is reconnecting the wallet but not sending the command or something like that.

**Comment by Dyrk** [comment]
I have to restart dcrstakepool 3-5 times per day.
On this hype number of registrations reached 1000 users and it goes down very often after new registration.

Currently, I'm making a bash script which will check page status and restart the service if it's not 200. But I hope we can find a better solution than restart and resync it so many times per day.

**Comment by jolan** [comment]
OK I'll give it another look soon (once the release work for 0.8.0 is done with).  I did look at it a couple weeks ago but had difficulty reproducing it on my unloaded testnet/dev pool which made it hard to debug.

**Comment by chappjc** [comment]
I'm running without the smarter connectivity PR changes, but when I tried it on testnet ages ago, I think the status URL triggered reconnect

**Comment by chappjc** [comment]
From an admin IP that is

**Comment by Dyrk** [comment]
Actually, it seems that this time it happened after failed `getstakeinfo` request.

What I see in the  dcrstakepool.log:
```
18:20:29 2017-02-08 [TRC] DCRS: Sending command [stakepooluserinfo] with id 6
18:20:31 2017-02-08 [ERR] DCRS: Websocket receive error from XXX:9110: unexpected EOF
18:20:31 2017-02-08 [TRC] DCRS: Disconnecting RPC client XXX:9110
18:20:31 2017-02-08 [TRC] DCRS: Shutting down RPC client XXX:9110
18:20:31 2017-02-08 [TRC] DCRS: RPC client input handler done for XXX:9110
18:20:31 2017-02-08 [TRC] DCRS: Sending command [stakepooluserinfo] with id 77
18:20:31 2017-02-08 [TRC] DCRS: RPC client output handler done for XXX:9110
```

And wallet.log:
```
18:20:32 2017-02-08 [WRN] RPCS: Websocket receive failed from client 81.2.253.91:45685: read tcp4 192.210.170.86:9110->81.2.253.91:45685: use of closed network connection

18:20:32 2017-02-08 [WRN] RPCS: Failed websocket send to client 81.2.253.91:45685: write tcp4 192.210.170.86:9110->81.2.253.91:45685: i/o timeout

18:20:32 2017-02-08 [INF] RPCS: Disconnected websocket client 81.2.253.91:45685
```

Wallet was online all this time and continue working ok. I see some voted tickets right after this sockets error. Of course, stakepool will never try to reconnect, so now only 2 wallets are still connected.

**Comment by chappjc** [comment]
Pretty sure going to /status from an admin IP is supposed to reconnect

**Comment by jholdstock** [comment]
@dajohi I believe this can be closed thanks to #494

**Comment by JoeGruffins** [comment]
@jholdstock ok to close this?
